### ISSUE TITLE
Problem with Object Detection in training

### ISSUE THREAD

**Issue by AkshajGupta** [ultralytics/ultralytics#3441]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

I am having trouble with the P R mAP50 and mAP50-95 values all being 0 when training

### Environment

- YoloV8
- Windows 10
- GPU TRX 1660

### Minimal Reproducible Example

I used this command
`yolo detect train data=config.yaml model=yolov8m.pt epochs=1 batch=4`

Here are my results

Transferred 469/475 items from pretrained weights
ClearML Task: created new task id=870ca4dadf1a47b1a8be4e5f0277a3db
2023-06-28 10:09:41,989 - clearml.Task - INFO - No repository found, storing script code instead
ClearML results page: https://app.clear.ml/projects/ba10c1f47b8d4c6ca95350cc78fdfff3/experiments/870ca4dadf1a47b1a8be4e5f0277a3db/output/log
ClearML Initialized a new task. If you want to run remotely, please add clearml-init and connect your arguments before initializing YOLO.
train: Scanning C:\Users\Level 42 AI User\Desktop\Medical_Dataset\labels\train.cache... 407 images, 7 backgrounds, 0 corrupt: 100%|██████████| 407/407 [00:00<?, ?it/s]
val: Scanning C:\Users\Level 42 AI User\Desktop\Medical_Dataset\labels\train.cache... 407 images, 7 backgrounds, 0 corrupt: 100%|██████████| 407/407 [00:00<?, ?it/s]
Plotting labels to runs\detect\train71\labels.jpg... 
optimizer: AdamW(lr=0.001, momentum=0.9) with parameter groups 77 weight(decay=0.0), 84 weight(decay=0.0005), 83 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 4 dataloader workers
Logging results to runs\detect\train71
Starting training for 1 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        1/1      3.56G      1.808       3.94      1.971          6        640: 100%|██████████| 102/102 [00:40<00:00,  2.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 51/51 [00:53<00:00,  1.05s/it]
                   all        407        562          0          0          0          0

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@AkshajGupta zero-mAP issues may indicate issues with AMP, PyTorch version, or CUDA/CUDNN drivers. I would recommend trying to train in one of our verified environments below for best compatibility.



## Environments

YOLOv8 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/quickstart) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

**Comment by AkshajGupta** [comment]
How do I use that to train my custom model? It says config.yaml is not detected? how do I add it to the server? @glenn-jocher

**Comment by glenn-jocher** [comment]
@AkshajGupta to train your custom model using YOLOv8, you need to provide a configuration file (config.yaml) that specifies the training settings and model architecture. If the config.yaml file is not detected, you can create one manually or use the default config.yaml provided in the YOLOv8 repository.

To add the config.yaml file to the server, you can simply place it in the same directory as your training script or command. Make sure that the file is correctly named and spelled as "config.yaml" for YOLOv8 to detect it properly.

Once you have the config.yaml file in the correct location, you can use it as an argument in your training command or script to specify the training configuration.

Please let me know if you have any further questions or need additional assistance.

**Comment by sech-ai** [comment]
como poder mejorar mi entrenamiento, ya que no me da bueno resultados. ¿Qué archivo debo editar?
![Captura](https://github.com/ultralytics/ultralytics/assets/132520086/629ba56b-6d4d-424c-8625-d2a213829f9e)

**Comment by glenn-jocher** [comment]
@sech-ai gracias por utilizar YOLOv8. Para mejorar los resultados de tu entrenamiento, es posible que desees revisar y ajustar algunos archivos de configuración.

Puedes editar el archivo `config.yaml` para modificar diferentes parámetros de entrenamiento, como la tasa de aprendizaje, el tamaño de lote (batch size), el número de épocas, entre otros. También puedes modificar la arquitectura del modelo en el archivo `models/yolov8.yaml`.

Te recomendaría revisar la documentación de YOLOv8 para obtener más detalles sobre los archivos de configuración y cómo ajustar los hiperparámetros para mejorar tu entrenamiento.

Espero que esta información te sea útil. Si tienes alguna otra pregunta o necesitas más ayuda, no dudes en preguntar. ¡Buena suerte con tu entrenamiento!
